src/sgl/func/tensor.cpp

```
@@ -0,0 +1,222 @@
+    size_t byte_offset = (m_desc.offset + start.value_or(0)) * el_stride;
+    return make_ref<BufferCursor>(m_desc.element_layout, m_storage, size, byte_offset);
+}
+
+void Tensor::view_inplace(slangpy::Shape shape, slangpy::Shape strides, int offset)
+{
+    SGL_CHECK(shape.valid(), "New shape must be valid");
+
+    if (!strides.valid() || strides.size() == 0)
+        strides = shape.calc_contiguous_strides();
+
+    for (size_t i = 0; i < strides.size(); ++i)
```

Cppcheck (2.20.0)

[style] The function 'grad_out' is never used.

(unusedFunction)